File: src/main/java/src/SistemaGestionEstudiante/Profesor.java
package SistemaGestionEstudiante;

import java.util.ArrayList;
import java.util.List;

public class Profesor {
    private String nombre;
    private List<Curso> cursos;

    public Profesor(String nombre) {
        this.nombre = nombre;
        this.cursos = new ArrayList<>();
    }

    public String getNombre() {
        return nombre;
    }

    public void asignarCurso(Curso curso) {
        cursos.add(curso);
    }

    public void asignarCalificacion(Estudiante estudiante, double calificacion) {
        estudiante.agregarCalificacion(calificacion);
    }

    @Override
    public String toString() {
        return "Profesor: " + nombre;
    }
}

File: src/main/kotlin/SistemaGestionEstudiante/Profesor.kt
package org.example.SistemaGestionEstudiante

class Profesor(val nombre: String) {
    private val cursos: MutableList<Curso>

    init {
        this.cursos = ArrayList<Curso>()
    }

    fun asignarCurso(curso: Curso) {
        cursos.add(curso)
    }

    fun asignarCalificacion(estudiante: Estudiante, calificacion: Double) {
        estudiante.agregarCalificacion(calificacion)
    }

    override fun toString(): String {
        return "Profesor: $nombre"
    }
}